index.bs

```diff
@@ -753,6 +753,62 @@ host it MUST perform the following steps for each script execution environment:
   };
 </pre>
 
+<div class="example">
+In this example we imagine a data logging device supporting up to 8 16-bit data
+channels. When data is sampled it is available on endpoint 1, which is a bulk IN
+endpoint. The maximum packet size on this endpoint is 16 bytes to support all 8
+channels being activated at the same time. To save bus bandwidth, however, any
+combination of channels can be activated and deactivated. The packet will only
+be the length necessary to transmit the data collected.
+
+To get started we open the device, select the first configuration (in case the
+operating system has not already done this during enumeration) and claim the
+data logging interface,
+
+<pre highlight="js">
+  await device.open();
+  if (device.configuration === null)
+    await device.setConfiguration(1);
+  await device.claimInterface(1);
+</pre>
+
+For this particular application we care about reading from channels 1, 2 and 5
+so we issue a control transfer to activate these channels,
+
+<pre highlight="js">
+  await device.controlTransferOut({
+      requestType: 'vendor',
+      recipient: 'interface',
+      request: 0x01,  // vendor-specific request: enable channels
+      value: 0x0013,  // 0b00010011 (channels 1, 2 and 5)
+      index: 0x0002   // Interface 1 is the recipient
+  });
+</pre>
+
+The application may now start polling the device for data. As we only expect
+data from 3 channels we request a 6 byte buffer. As long as we receive a
+complete buffer the captured values are printed to the console log. If the
+device has encountered an error and signals this by stalling the endpoint then
+the error is cleared before continuing,
+
+<pre highlight="js">
+  while (true) {
+    let result = await data.transferIn(1, 6);
+
+    if (result.data && result.data.byteLength === 6) {
+      console.log('Channel 1: ' + result.data.getUint16(0));
+      console.log('Channel 2: ' + result.data.getUint16(1));
```

Comment:
If `data` is a `DataView`, you need `getUint16(2)`.

I don't think you specified anywhere that the data was big-endian. Is that just standard for USB?